cmd/codegen/generator/typescript/templates/functions.go

```diff
@@ -301,14 +312,51 @@ func (funcs typescriptTemplateFuncs) sortInputFields(s []introspection.InputValu
 
 func (funcs typescriptTemplateFuncs) sortEnumFields(s []introspection.EnumValue) []introspection.EnumValue {
 	slices.SortStableFunc(s, func(x, y introspection.EnumValue) int {
-		return cmp.Compare(funcs.formatEnum(x.Name), funcs.formatEnum(y.Name))
+		return cmp.Compare(strcase.ToCamel(x.Name), strcase.ToCamel(y.Name))
```

Comment:
Important! Make sure to call `slices.Clone` before doing any of this.

I had to add this for go, should have added it here too - but without this, we're actually mutating the original (which led to some surprising results when I did this)